src/accelerate/utils/modeling.py

```diff
@@ -1311,7 +1313,7 @@ def get_state_dict_offloaded_model(model: nn.Module):
 def load_checkpoint_in_model(
     model: nn.Module,
     checkpoint: Union[str, os.PathLike],
-    device_map: Optional[Dict[str, Union[int, str, torch.device]]] = None,
```

Comment:
This typing is wrong as `None` is not an accepted value in this function (raises an error at https://github.com/huggingface/accelerate/blob/bd72a5f1a80d5146554458823f8aeda0a9db5297/src/accelerate/utils/modeling.py#L436)

Reply:
I leave it as is because changing to `     device_map: Dict[str, Union[int, str, torch.device]]` raises errors in the CI.

Reply:
Thx for noticing this. This should be fixed [here](https://github.com/huggingface/accelerate/pull/2218#issue-2026649653).